src/Configuration/Entry/Badge.php

```diff
@@ -35,20 +35,45 @@
 
 namespace Infection\Configuration\Entry;
 
+use InvalidArgumentException;
+use Safe\Exceptions\PcreException;
+use function Safe\preg_match;
+use function Safe\sprintf;
+
 /**
  * @internal
  */
 final class Badge
 {
-    private string $branch;
+    private ?string $exactBranchMatch;
+    private ?string $matchBranchRegex;
+
+    public function __construct(?string $exactBranchMatch, ?string $matchBranchRegex)
+    {
+        if ($matchBranchRegex !== null) {
+            try {
+                // Yes, the `@` is intentional. For some reason, `thecodingmachine/safe` does not suppress the warnings here
```

Comment:
Really had to take a shower after this: I was hoping that `thecodingmachine/safe` did deal with this stuff, but nope.

Reply:
Looked into this: `"format": "regex"` would collide with simple values